In [7]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# HW 1

In [8]:
# Данные
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

# Шаг 1: Ковариация вручную
mean_zp = np.mean(zp)
mean_ks = np.mean(ks)
cov_manual = np.mean((zp - mean_zp) * (ks - mean_ks))

# Шаг 2: Ковариация с помощью numpy
cov_np = np.cov(zp, ks, ddof=0)[0, 1]

# Шаг 3: Коэффициент корреляции Пирсона вручную
std_zp = np.std(zp, ddof=0)
std_ks = np.std(ks, ddof=0)
pearson_corr_manual = cov_manual / (std_zp * std_ks)

# Шаг 4: Коэффициент корреляции Пирсона с использованием numpy и pandas
pearson_corr_np = np.corrcoef(zp, ks)[0, 1]
pearson_corr_pd = pd.Series(zp).corr(pd.Series(ks))

{
    "cov_manual": cov_manual,
    "cov_np": cov_np,
    "pearson_corr_manual": pearson_corr_manual,
    "pearson_corr_np": pearson_corr_np,
    "pearson_corr_pd": pearson_corr_pd
}

{'cov_manual': 9157.84,
 'cov_np': 9157.84,
 'pearson_corr_manual': 0.8874900920739162,
 'pearson_corr_np': 0.8874900920739162,
 'pearson_corr_pd': 0.8874900920739162}

### Ковариация:
- Рассчитанная вручную: 9157.84
- Рассчитанная с помощью функции cov из numpy: 9157.84
### Коэффициент корреляции Пирсона:
- Рассчитанный вручную: 0.8875
- Рассчитанный с помощью numpy: 0.8875
- Рассчитанный с помощью pandas: 0.8875

# HW 2

### Для расчета доверительного интервала используем t-распределение, так как у нас небольшая выборка и неизвестна дисперсия генеральной совокупности.

In [9]:
# Данные
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

# Параметры выборки
mean_iq = np.mean(iq)
std_iq = np.std(iq, ddof=1)
n = len(iq)
confidence = 0.95

# Доверительный интервал
t_critical = stats.t.ppf((1 + confidence) / 2, df=n-1)
margin_of_error = t_critical * (std_iq / np.sqrt(n))
confidence_interval_iq = (mean_iq - margin_of_error, mean_iq + margin_of_error)

confidence_interval_iq

(110.55608365121006, 125.64391634878993)

# HW 3

Для расчета доверительного интервала используем z-распределение, так как дисперсия генеральной совокупности известна.

In [10]:
# Данные
variance_population = 25
n_footballers = 27
mean_sample = 174.2
confidence = 0.95

# Доверительный интервал
z_critical = stats.norm.ppf((1 + confidence) / 2)
std_error = np.sqrt(variance_population / n_footballers)
margin_of_error = z_critical * std_error
confidence_interval_height = (mean_sample - margin_of_error, mean_sample + margin_of_error)

confidence_interval_height

(172.3140237765397, 176.08597622346028)